projects/rocprofiler-sdk/source/lib/rocprofiler-sdk-attach/attach.cpp

```diff
@@ -0,0 +1,90 @@
+// MIT License
+//
+// Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#include "attach.h"
+#include "code_object_registration.hpp"
+#include "queue_registration.hpp"
+#include "table.hpp"
+
+#include "lib/common/logging.hpp"
+
+void
+init_logging()
+{
+    rocprofiler::common::init_logging("ROCPROFILER_ATTACH");
+}
+
+// ensure that logging is always initialized when library is loaded
+bool init_logging_at_load = (init_logging(), true);
+
+using rocprofiler_register_functor_t = void (*)(uint64_t, void*, uint64_t);
+
+ROCPROFILER_EXTERN_C_INIT
+
+int
+rocprofiler_attach_set_api_table(const char* name,
+                                 uint64_t    lib_version,
+                                 uint64_t    lib_instance,
+                                 void**      tables,
+                                 uint64_t    num_tables,
+                                 void*       register_functor)
```

Comment:
I would prefer the `register_functor` to be `decltype(::rocprofiler_set_api_table)*` -- i.e. function pointer to function with signature identical to `rocprofiler_set_api_table`